hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -979,6 +995,11 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to reconcile machine approver: %w", err)
 	}
 
+	// Reconcile the AWS OIDC discovery
```

Comment:
I see the func returns nil if hcluster.Spec.Platform.Type is not aws but (at least for now) we might want to be consistent with the other platform specific reconcilers and gate the call with
```
	switch hcluster.Spec.Platform.Type {
	case hyperv1.AWSPlatform:
```

Reply:
Updated it, although I find it slightly silly to have a one-entry switch statement

Reply:
yeh I agree, though I think being consistent with the pattern would make adding more platforms and refactor easier